pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunction.java

```diff
@@ -75,6 +94,20 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
       Map<ExpressionContext, BlockValSet> blockValSetMap) {
     if (blockValSetMap.isEmpty()) {
       aggregationResultHolder.setValue(aggregationResultHolder.getDoubleResult() + length);
+    } else if (!blockValSetMap.containsKey(STAR_TREE_COUNT_STAR_EXPRESSION)) {
+      if (!_nullHandlingEnabled) {
```

Comment:
Could this branch be hit?

Reply:
Suggest changing the branching logic as following for clarity:
```
if (blockValSetMap.isEmpty()) {
  ...
} else if (_nullHandlingEnabled) {
  ...
} else {
  // Star-tree pre-aggregated values
  ...
}

Reply:
I still need to check for: `!blockValSetMap.containsKey(STAR_TREE_COUNT_STAR_EXPRESSION)`

Reply:
Updated for readability

Reply:
@Jackie-Jiang We have three branches:
1- `blockValSetMap.isEmpty()` for `count(*)`, and no StarTree.
2- `!blockValSetMap.containsKey(STAR_TREE_COUNT_STAR_EXPRESSION)` for count(columnName), and no StarTree (please see the `!` at the beginning). Example: `count(bigDecimalColumn)`. Note: we can have count(bigDecimalColumn) and nullHandling is either enable or not enabled.
3- StarTree.